src/SnapshotManager.ts

```diff
@@ -471,6 +481,17 @@ export class SnapshotManager {
       instructions.drop.push(mapping.fields[property].joinTable.name);
     }
 
+    function createForeign(mapping, property, targetMapping) {
```

Comment:
Can't we reuse an existing method somewhere? It looks like something else in the file.